tests/core/test_snapshot.py

```diff
@@ -296,10 +296,10 @@ def test_missing_intervals_partial(make_snapshot):
     start = "2023-01-01"
     end_ts = to_timestamp(start) + 1000
     assert snapshot.missing_intervals(start, end_ts) == [
-        (to_timestamp(start), to_timestamp("2023-01-02")),
+        (to_timestamp(start), end_ts),
```

Comment:
This doesn't look right to me either. The missing intervals should be aligned with the model's interval unit, which is not the case here. Eg. for a daily model interval that is 1 second long is meaningless.